test/string.spec.ts

```diff
@@ -0,0 +1,485 @@
+import { describe, expect, it } from 'vitest';
+import { faker, FakerError } from '../src';
+import { seededTests } from './support/seededRuns';
+import { times } from './support/times';
+
+const NON_SEEDED_BASED_RUN = 5;
+
+describe('string', () => {
+  seededTests(faker, 'string', (t) => {
+    t.describe('alpha', (t) => {
+      t.it('noArgs')
+        .itRepeated('repeated', 5, 5)
+        .it('with length', { length: 6 })
+        .it('with casing', { casing: 'lower' })
+        .it('with casing', { casing: 'upper' })
+        .it('with casing', { casing: 'mixed' })
+        .it('with bannedChars', { bannedChars: 'abcdefghijk' })
+        .it('with length, casing and bannedChars', {
+          length: 7,
+          casing: 'lower',
+          bannedChars: 'lmnopqrstu',
+        });
+    });
+
+    t.describe('alphanumeric', (t) => {
+      t.it('noArgs')
+        .itRepeated('repeated', 5, 5)
+        .it('with length', { length: 6 })
+        .it('with casing', { casing: 'lower' })
+        .it('with casing', { casing: 'upper' })
+        .it('with casing', { casing: 'mixed' })
+        .it('with bannedChars', { bannedChars: 'abcdefghijk12345' })
+        .it('with length, casing and bannedChars', {
+          length: 7,
+          casing: 'lower',
+          bannedChars: 'lmnopqrstu67890',
+        });
+    });
+
+    t.describe('hexadecimal', (t) => {
+      t.it('noArgs')
+        .it('with length', { length: 6 })
```

Comment:
Why no length parameter check here?

```suggestion
        .itRepeated('with length parameter', 5, 5)
        .it('with length', { length: 6 })
```

Reply:
`hexadecimal` doesn't have a only number option => also moved to #1349